### PR TITLE
 [StyleCop] Fix all remaining warnings in all C# projects

### PR DESCRIPTION
**Pull request by Aliandi**
Fix StyleCop warnings left to make #1425 pass and enforce StyleCop.

- Reorder methods
- Reorder usings
- Create new files for classes
- Reorder usings
- Fix encoding

### CODE REVIEW

**Review by gasper-az — APPROVED** [review]
LGTM

**Review by denscollo — APPROVED** [review]
LGTM 👍

File: .NET/Microsoft.Recognizers.Definitions/Microsoft.Recognizers.Definitions.csproj [hunk 1]
```diff
@@ -12,7 +12,17 @@
...
   </PropertyGroup>
+<PropertyGroup>
+	  <!--
+		Make sure any documentation comments which are included in code get checked for syntax during the build, but do
+		not report warnings for missing comments.
 
+		CS1573: Parameter 'parameter' has no matching param tag in the XML comment for 'parameter' (but other parameters do)
+		CS1591: Missing XML comment for publicly visible type or member 'Type_or_Member'
+	  -->
+	  <DocumentationFile>$(OutputPath)$(AssemblyName).xml</DocumentationFile>
```
Comment (JuanAr) [inline .NET/Microsoft.Recognizers.Definitions/Microsoft.Recognizers.Definitions.csproj:23]: Is this file missing? I know it is empty now, but as far as I can recall, we included an empty xml file in other similar PRs

**Review by JuanAr — COMMENTED** [review]
One comment regarding the XML comment file.

File: .NET/Microsoft.Recognizers.Definitions/Microsoft.Recognizers.Definitions.csproj (hunk 1 above)
Comment (Aliandi) [reply .NET/Microsoft.Recognizers.Definitions/Microsoft.Recognizers.Definitions.csproj:23]: Yup, I missed it. Added now. Tysm!

**Review by JuanAr — APPROVED** [review]
👍

**Comment by denscollo** [comment]
Replaced by #1458

**Pull request closed without merging**